frame/authority-discovery/src/lib.rs

```diff
@@ -106,17 +109,15 @@ impl<T: Config> Pallet<T> {
 	fn initialize_keys(keys: &Vec<AuthorityId>) {
 		if !keys.is_empty() {
 			assert!(Keys::<T>::get().is_empty(), "Keys are already initialized!");
-			let bounded_keys = Self::to_bounded_vec((*keys).clone());
+
+			let bounded_keys =
+				BoundedVec::<AuthorityId, T::MaxAuthorities>::try_from((*keys).clone())
+					.expect("Keys vec too big");
```

Comment:
expect is a panic.

you need to properly handle the result

Reply:
This method already panics as it's in the initialisation stage of the runtime.